src/main.cpp

```diff
@@ -3519,13 +3695,15 @@ bool SignBlock(CBlock& block, CWallet& wallet, CAmount nFees)
     CKey key;
     CMutableTransaction txCoinStake;
     CTransaction txNew;
+    int nBestHeight = chainActive.Tip()->nHeight;
     txCoinStake.nTime &= ~STAKE_TIMESTAMP_MASK;
 
     int64_t nSearchTime = txCoinStake.nTime; //! search to current time
     if (fDebug)
-        LogPrintf("DEBUG : nSearchTime=%i, nLastCoinStakeSearchTime=$i\n", nSearchTime, nLastCoinStakeSearchTime);
+        LogPrintf("DEBUG : nSearchTime=%i, nLastCoinStakeSearchTime=%i\n", nSearchTime, nLastCoinStakeSearchTime);
     if (nSearchTime > nLastCoinStakeSearchTime) {
-        if (wallet.CreateCoinStake(wallet, block.nBits, nSearchTime-nLastCoinStakeSearchTime, nFees, txCoinStake, key)) {
+        int64_t nSearchInterval = nBestHeight+1 > 0 ? 1 : nSearchTime - nLastCoinStakeSearchTime;
```

Comment:
nBestHeight+1 will always be greater than 0 so nSearchInterval will always be 1?

Reply:
Removed and rebased it.